src/index.js

```diff
@@ -144,7 +151,13 @@ export class WT {
       delete this.loaderImage.onload;
       reject();
     };
-    this.loaderImage.src = `${this.getUrl()}?${query}`;
+    const params = new this.context.URLSearchParams(query);
+    const sendBeaconResult = this.context.navigator.sendBeacon(`${this.getBeaconUrl()}`, params);
```

Comment:
afaict, the only time this would return false is if the data limit is surpassed, see the note under **Return Value** [here](https://www.w3.org/TR/beacon/#dom-navigator-sendbeacon). given that, I'm a bit skeptical that we'd ever get false returned but I think we don't know for sure. that being said, can we just hit the new POST endpoint instead (via a standard AJAX call) so we don't have to maintain the `track.gif` endpoint? then, we also wouldn't need `beaconUrl` and `trackerUrl`, we can just have the latter.

additionally, not all browsers will have `navigator.sendBeacon`, so I think it would be wise to add a polyfill to avoid crashing in this case, or checking for the existence of the function and falling back to an AJAX call if it doesn't exist.

Reply:
@bcharna i believe there's an example of hitting the data limit (via an overly long query string) in the PR description. 

that said, i think the only reason that's likely to happen is due to the how we batch events together. i'm not sure if there's any reason to do that with the beacon API, but if we _can_ remove that it would be a win in terms of complexity.

re: polyfilling, i think we can safely offload that to consumers of the library rather than make that decision for everybody.

Reply:
yeah, except the data limits may be different.

also agree batching may not be necessary, but likely something we can punt on for now to keep change size smaller.

fair point on polyfilling – are you proposing we go with my latter suggestion (check for existence of function and hit AJAX if not present)?

Reply:
proposing we leave as is + add a note in the readme that `sendBeacon` support is required and users may polyfill if neccesary.

Reply:
ah, I see. sorry, I sort of combined two pieces of feedback in the original comment. so I think to summarize we're saying:

1. end users of wt (for us, landing, platform and today) will polyfill themselves.
2. we'll make a POST request instead of an image request in the case that `this.context.navigator.sendBeacon` returns `false`

Reply:
Do we want to completely get rid of the image request (i.e. try sendBeacon then try POST)? or try all three (i.e. try sendBeacon then try POST then try image)?

Reply:
I don't see a reason to keep the image request around – all browsers should support POST via standard AJAX. but to clarify, @nopelluhh was proposing only to fallback to the standard AJAX if `sendBeacon` returns false, thus we assume that the function is present (this means we'll want to add a polyfill in our apps that use wt).

Reply:
What are your thoughts on a [sendbeacon-polyfill](https://www.npmjs.com/package/sendbeacon-polyfill)? I see from the [source code ](https://github.com/kronicker/sendbeacon-polyfill/blob/master/sendbeacon.js) that it already handles sending as AJAX

Reply:
it looks like that's a fork of https://github.com/miguelmota/Navigator.sendBeacon, can we see what the difference is between those? it looks like the original repo has more stars/has been updated more recently but have never worked w/ either.

and yep, a lot of cleanup is possible here!